BioCro/DARPA/set_MA_trt.R

```diff
@@ -0,0 +1,108 @@
+## Function that creates a link table between treatments and their RE indices 
+## And produces a list of mcmc.list objects for only the treatment-level posteriors
+## Target site/treatment is re-labeled "beta.o" so additional PEcAn functions will work 
+
+
+set_MA_trt <- function(settings){
+  
+  library(coda)
+  
+  #obtain dbfiles path
+  con <- PEcAn.DB::db.open(settings$database$bety)
+  postid <- settings$pfts$pft$posteriorid
+  fname <- PEcAn.DB::dbfile.file(type = "Posterior", id = postid, con = con)
+  fpath <- gsub(paste0(postid, "\\/.*"), postid, fname)
+
+  #load jagged.data and trait.mcmc
+  load(file.path(fpath, "jagged.data.Rdata"))
+  load(file.path(fpath, "trait.mcmc.Rdata"))
+
+  #convert jagged.data into match table, only return traits for which there is an associated mcmc output
+  trt.match <- lapply(jagged.data, collapse)[names(trait.mcmc)]
+  
+  #save trt.match, 2 places
+  save(trt.match, file = file.path(settings$database$dbfiles, "posterior", postid, "trt.match.Rdata"))
+  save(trt.match, file = file.path(settings$pfts$pft$outdir, "trt.match.Rdata"))
+
+  
+  #save existing trait.mcmc with different name, 2 places
+  save(trait.mcmc, file = file.path(settings$database$dbfiles, "posterior", postid, "trait.mcmc.original.Rdata"))
+  save(trait.mcmc, file = file.path(settings$pfts$pft$outdir, "trait.mcmc.original.Rdata"))
+
+  #create new trait.mcmc of combined random effects
+  new.trait.mcmc <- mapply(FUN = RE_combine, mc = trait.mcmc, trt = trt.match, 
+                           SIMPLIFY = FALSE)
+  
+  #identify target site and treatment
+  target.site <- settings$run$site$id
+  target.trt <- settings$meta.analysis$treatment
+
+  #label target site/treatment "beta.o"
+  final.trait.mcmc <- mapply(FUN = rename_cols, trait.mcmc = new.trait.mcmc, trt = trt.match,
+                             MoreArgs = list(target.site = target.site, target.trt = target.trt),
+                             SIMPLIFY = FALSE)
+  
+  #rename as trait.mcmc and save as "trait.mcmc.Rdata", 2 places
+  trait.mcmc <- final.trait.mcmc
```

Comment:
This renaming step seems possibly unnecessary? `final.trait.mcmc` could just directly be saved as "trait.mcmc.Rdata" in lines 47 and 48.

Reply:
Hmm yes, but then when loaded in for the next step, it will be named final.trait.mcmc. I thought maybe the next function would load the Rdata object and then assume it was named trait.mcmc

Reply:
Yeah, you're right about that, the .Rdata file needs to be named trait.mcmc. But line 47 could be `save(final.trait.mcmc, file = file.path(settings$database$dbfiles, "posterior", postid, "trait.mcmc.Rdata"))`, right. That would then make line 46 unnecessary.

Reply:
When the .Rdata object loads, it will retain the object name it was saved as (I'm pretty sure). So then the next step will load in the file name "trait.mcmc.Rdata", but the object name will still be "final.trait.mcmc". So if the function requires the object name to be trati.mcmc, it still won't locate it.

Reply:
Ah, okay. I'm not as familiar with .Rdata files, so some of their behavior seems odd to me.